include/spblas/vendor/onemkl_sycl/mkl_allocator.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <sycl.hpp>
+
+namespace spblas {
+namespace mkl {
+
+template <typename T, std::size_t Alignment = 0>
+class mkl_allocator {
+public:
+  using value_type = T;
+  using pointer = T*;
+  using const_pointer = const T*;
+  using reference = T&;
+  using const_reference = const T&;
+  using size_type = std::size_t;
+  using difference_type = std::ptrdiff_t;
```

Comment:
why do we have a `typename T` on the allocator?  don't we have potential for creating all sorts of kinds of things ?

Reply:
It currently follows the std::allocator design. In the operation, we will use `allocator<char>` to create the workspace.